### PR TITLE
Added retry action for CI

### PR DESCRIPTION
**Pull request by agunapal**
## Description

The CI runs can be flaky because of network issues. 
Added a retry action to run the sanity script with the following properties

- Timeout = 60 minutes
- Max Retries = 3

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

## Feature/Issue validation/testing

Please describe the Unit or Integration tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
Logs for Test A

- [ ] Test B
Logs for Test B

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: .github/workflows/ci_cpu.yml [hunk 1]
```diff
@@ -36,8 +36,12 @@ jobs:
         run: |
           python ts_scripts/install_dependencies.py --environment=dev
       - name: Torchserve Sanity
-        run: |
-          python torchserve_sanity.py
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 60
```
Comment (msaroufim) [inline .github/workflows/ci_cpu.yml:41]: Does the timeout apply per run or for all runs?

File: .github/workflows/ci_cpu.yml [hunk 2]
```diff
@@ -36,8 +36,12 @@ jobs:
...
+          max_attempts: 3
```
Comment (msaroufim) [inline .github/workflows/ci_cpu.yml:42]: Strange how the Mac and Ubuntu time are now taking over an hour, so there's a retry happening but no failures?

Comment (agunapal) [reply .github/workflows/ci_cpu.yml:42]: Here it seems like its 60 minutes for each run and then it timeouts and tries again. I'm not sure why each run is taking more than 60 minutes. As per previous runs, its around 25 minutes

File: .github/workflows/ci_cpu.yml (hunk 1 above)
Comment (agunapal) [reply .github/workflows/ci_cpu.yml:41]: Its 60 minutes per run

File: .github/workflows/ci_cpu.yml (hunk 2 above)
Comment (msaroufim) [inline .github/workflows/ci_cpu.yml:42]:
```suggestion
          max_attempts: 3
          retry_on: error
          
```

Comment (agunapal) [reply .github/workflows/ci_cpu.yml:42]: Thanks @msaroufim . This might be the issue

**Review by msaroufim — APPROVED** [review]

**Comment by msaroufim** [comment]
@agunapal what do you think should we merge this?

**Review by mreso — APPROVED** [review]

**Pull request merged**